python/gtsam/examples/IMUKittiExampleGPS.py

```diff
@@ -0,0 +1,332 @@
+"""
+Example of application of ISAM2 for GPS-aided navigation on the KITTI VISION BENCHMARK SUITE
+
+Author: Varun Agrawal
+"""
+import argparse
+from typing import List
+
+import gtsam
+import numpy as np
+from gtsam import ISAM2, Point3, Pose3, noiseModel
+from gtsam.symbol_shorthand import B, V, X
+
+
+class KittiCalibration:
+    """Class to hold KITTI calibration info."""
+    def __init__(self, body_ptx: float, body_pty: float, body_ptz: float,
+                 body_prx: float, body_pry: float, body_prz: float,
+                 accelerometer_sigma: float, gyroscope_sigma: float,
+                 integration_sigma: float, accelerometer_bias_sigma: float,
+                 gyroscope_bias_sigma: float, average_delta_t: float):
+        self.bodyTimu = Pose3(gtsam.Rot3.RzRyRx(body_prx, body_pry, body_prz),
+                              gtsam.Point3(body_ptx, body_pty, body_ptz))
+        self.accelerometer_sigma = accelerometer_sigma
+        self.gyroscope_sigma = gyroscope_sigma
+        self.integration_sigma = integration_sigma
+        self.accelerometer_bias_sigma = accelerometer_bias_sigma
+        self.gyroscope_bias_sigma = gyroscope_bias_sigma
+        self.average_delta_t = average_delta_t
+
+
+class ImuMeasurement:
+    """An instance of an IMU measurement."""
+    def __init__(self, time: float, dt: float, accelerometer: gtsam.Point3,
+                 gyroscope: gtsam.Point3):
+        self.time = time
+        self.dt = dt
+        self.accelerometer = accelerometer
+        self.gyroscope = gyroscope
+
+
+class GpsMeasurement:
+    """An instance of a GPS measurement."""
+    def __init__(self, time: float, position: gtsam.Point3):
```

Comment:
same story here -- prefer a dataclass

Reply:
Python 3.6 is end of life in 2 months though -- https://devguide.python.org/#status-of-python-branches
https://endoflife.date/python

I think security fixes aren't supported after that date, so it could be insecure to use. Any reason to keep supporting 3.6?

Reply:
The GTSAM user base. Some of them are companies (which take forever to update their software infra) and we would need Frank's express approval to make this update.

But anyway, this is not directly related to the functionality of this PR (more of a nice-to-have) so we can discuss this later. :slightly_smiling_face: